src/qibo/abstractions/callbacks.py

```diff
@@ -97,7 +77,7 @@ def nqubits(self, n: int):
         if self.partition is None: # pragma: no cover
             self.partition = list(range(n // 2 + n % 2))
 
-        if len(self.partition) < n // 2:
+        if len(self.partition) <= n // 2:
```

Comment:
Does this affects some of our results?

Reply:
This should not affect results when the entanglement entropy is concerned. This is because of the entropy's property that it doesn't matter which bi-partition we trace out in order to calculate it. The idea of this `if` is to always trace out the larger bi-partition so that we then calculate `log(rho)` (which requires diagonalization) for a smaller matrix.

For example if we have 10 qubits and the bipartition is [0, 1] and [2, 3, ..., 9], we can either trace out [0, 1] and diagonalize a 2^8 x 2^8 matrix or trace out [2, 3, ..., 9] and diagonalize a 4x4 matrix. Clearly the latter is more efficient. The reason I put the equal here is so that we flip to the smaller bi-partition when the number of qubits is odd. For example for 5 qubits [0, 1, 2] should be flipped to [3, 4] so that the diagonalization happens for one qubit less (it is a minor change).